### PR TITLE
[#8249] "" when grant access to ens dapp shown, it prevents the name …

### LINKED ISSUE #8249: "" when grant access to ens dapp shown, it prevents the name to be registered
# Problem
Can't register ens username . 

## Steps

1. open status on mainnet
1. open names.statusnet.eth dapp
1. select any free username and tap Grant access -> note that "" shown instead of dapp name
image
1. tap Allow -> Register -> Sent SNT

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #8249

### CODE REVIEW

File: src/status_im/browser/core.cljs [hunk 1]
```diff
@@ -179,7 +179,7 @@
                                assoc
                                :url (str gateway path)
                                :resolving? false)
-                       (assoc-in [:browser/options :resolved-ens host] gateway))})))
+                       (assoc-in [:browser/options :resolved-ens host] (subs gateway 8)))})))
```
Comment (jeluard) [inline src/status_im/browser/core.cljs]: Magic number? Can we use a separate function for that?

**Review by jeluard — CHANGES_REQUESTED** [review]

**Review by yenda — APPROVED** [review]

**Comment by flexsurfer** [comment]
@jeluard done

File: src/status_im/browser/core.cljs (hunk 1 above)
Comment (yenda) [reply src/status_im/browser/core.cljs]: it is not a real fix anyway

**Comment by jeluard** [comment]
@flexsurfer Thanks!

**Review by jeluard — APPROVED** [review]

File: src/status_im/browser/core.cljs [hunk 2]
```diff
@@ -179,7 +179,7 @@
                                assoc
                                :url (str gateway path)
                                :resolving? false)
-                       (assoc-in [:browser/options :resolved-ens host] gateway))})))
+                       (assoc-in [:browser/options :resolved-ens host] (string/replace gateway #"https://" "")))})))
```
Comment (yenda) [inline src/status_im/browser/core.cljs]: nah if you are going to remove the https prefix just don't put it there in the first place

`(str "https://" gateway path)`

**Review by yenda — CHANGES_REQUESTED** [review]
don't add https:// in the first place, also it is ok for a hotfix but I think there is more to it that this

**Comment by flexsurfer** [comment]
@yenda done

**Review by yenda — APPROVED** [review]

**Comment by churik** [comment]
Checked on OS 11.4.1 (IPhone 7), Android 8 (LG V20):
- can grant access to `names .statusnet.eth`
- can grant access to other apps (on Mainnet and Ropsten)
- can revoke and grant access again

**Pull request merged**